src/megaapi_impl.cpp

```diff
@@ -3832,6 +3837,8 @@ void MegaApiImpl::init(MegaApi *api, const char *appKey, MegaGfxProcessor* proce
     syncLowerSizeLimit = 0;
     syncUpperSizeLimit = 0;
 
+    addLoggerClass(initLogger);
```

Comment:
actually this does not make much sense. that function is static. You can simply use the static `MegaApi::addLoggerObject` before creating any MegaApi object. I've already updated `develop` branch in MEGAcmd project to proceed that way.

Reply:
Agreed, I should have seen that the loggers were ultimately static.  I've undone those changes.  thanks!